lib/galaxy/webapps/galaxy/controllers/user.py

```diff
@@ -570,12 +572,30 @@ def __validate_login(self, trans, **kwd):
                 status = 'warning'
         return (message, status, user, success)
 
+    def check_user_library_import_dir(self, user):
+        if self.app.config.user_library_import_dir_auto_creation:
```

Comment:
Thanks for the contribution!

I think this PR is causing the Tool Shed tests to fail because it reuses this same controller with a different ``app`` and ``config`` structure. Can you move this logic into ``trans.handle_user_login`` (lib/galaxy/web/framework/webapp.py) instead of placing it in the controller. For many reasons we like to keep controllers as slim as possible so I think it would be a better place for it anyway. If that doesn't fix the TS tests you could then use ``getattr(app.config, "user_library_import_dir_auto_creation", False)`` to access the config in a way that would harmlessly cause a no-op from within the tool shed.